test/test_proxy_tensor.py

```diff
@@ -241,18 +254,26 @@ def f2_logging_tensor(x):
             self.assertTrue(is_any_digamma(traced))
 
     @unittest.skipIf(not USE_TORCHVISION, "test requires torchvision")
-    def test_resnet18_backward_trace(self, device):
+    def test_resnet18_backward_trace(self):
         mod = torchvision.models.resnet18()
 
-        def f(x):
-            for a in mod.parameters():
-                a.grad = None
-            out = mod(x)
-            out.sum().backward()
-            return [a.grad for a in mod.parameters()]
-
-        inp = torch.randn(3, 3, 250, 250, requires_grad=True)
-        self._test(f, [inp])
+        # An old version of this test called the module directly.  This works
+        # for tracing_mode == "real", but for fake tensors, we also have to
+        # ensure that the parameters and buffers get wrapped in fake tensors
+        # because free fake tensors are not supported.  Fortunately stateless
```

Comment:
you can also do: 
```
        mode = FakeTensorMode(inner=None)
        with torch._subclasses.fake_tensor.FakeCopyMode(mode):
            mod_copied = copy.deepcopy(mod)
```

to convert the buffers/parameters (plus tensors which are neither)

Reply:
wow, when did we add FakeCopyMode? Is it only supposed to be used for this?

Reply:
I think you were on [PTO](https://github.com/pytorch/pytorch/pull/79580)... Yea it's only supposed to be used for deepcopy. Eventually it might be nice to merge with FakeTensorMode but I think there were some issues at the time with inheriting from both `TorchFunctionMode` and `TorchDispatchMode`.

Reply:
FakeTensorMode needs to be converted into a synthetic class that contains the function and dispatch mode, probably.